File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/Ledger.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import java.util.List;
import lombok.Builder;

/**
 * A folio/invoice breakdown with a total: {@code lines} ({@link LedgerLine}) rendered as
 * concept-left, monospace-amount-right rows (negative amounts in error red, included lines show
 * their label instead of an amount), then a divider and the {@code totalLabel} with the big {@code
 * total} ({@code null} → the client computes the sum of non-included amounts). Amounts are
 * formatted client-side with the {@code currency} symbol. Read-only, no actions. Design-system
 * neutral, dark-mode aware.
 */
@Builder
public record Ledger(
    String id,
    String currency,
    String totalLabel,
    List<LedgerLine> lines,
    Double total,
    String style,
    String cssClasses)
    implements Component {}

File: frontend/web/monorepo/libs/mateu/src/mateu/shared/apiClients/dtos/componentmetadata/Ledger.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";
import LedgerLine from "@mateu/shared/apiClients/dtos/componentmetadata/LedgerLine";

export default interface Ledger extends ComponentMetadata {
    currency?: string
    totalLabel?: string
    lines?: LedgerLine[]
    total?: number
}
